### PR TITLE
Feature/2733 show stock available in error msg

### LINKED ISSUE #2733: Showing number of available products
## What is the motivation for adding / enhancing this feature?

When a customer wants to order several pieces of a given product, for ex 10, and this quantity is not available - he must try one by one to determine how many items he can order. 

Showingnumberofavailableprod

In the message about the lack of a product:
_"We don't have as many "product-name" as you requested"_
we can add the information _"We currently have: (available quantity)"_.
Thanks to this, the customer will know how many pieces he can order

## What are the acceptance criteria 

- [ ] When I try to add an unavailable quantity of product to the shopping cart, a message appears: _"We don't have as many "product-name" as you requested. We currently have: (available quantity)"_.

## Can you complete this feature request by yourself?

- [ ] YES
- [x] NO

## Which [Release Cycle](https://docs.vuestorefront.io/guide/basics/release-cycle.html) state this refers to? Info for developer.
Pick one option.

- [x] This is a normal feature request. This should be available on https://test.storefrontcloud.io and then after tests this can be added to next Vue Storefront version. In this case Developer should create branch from `develop` branch and create Pull Request `2. Feature / Improvement` back to `develop`.
- [ ] (**Pick this option only if you're sure**) This is an important improvement request for current Release Candidate version on https://next.storefrontcloud.io and should be placed in next RC version. In this case Developer should create branch from `release` branch and create Pull Request `3. Stabilisation fix` back to `release`.
- [ ] (**Pick this option only if you're sure**) This is a critical improvement request for current Stable version on https://demo.storefrontcloud.io and should be placed in next stable version. In this case Developer should create branch from `hotfix` or `master` branch and create Pull Request `4. Hotfix` back to `hotfix`.

### PR DESCRIPTION
**Pull request by collymore**
### Related issues

closes #2733

### Short description and why it's useful
It displays the stock available if the user tries to add more then is available to their cart. 

It takes into account the total already in the cart and available stock and works out if the requested amount can be added. 

The user can now see the stock available in the notification and reduce their requested amount so they can add the product to cart. 

### Screenshots of visual changes before/after (if there are any)
stock-notice

### Which environment this relates to
Check your case. In case of any doubts please read about [Release Cycle](https://docs.vuestorefront.io/guide/basics/release-cycle.html)

- [x] Test version (https://test.storefrontcloud.io) - this is a new feature or improvement for Vue Storefront. I've created branch from `develop` branch and want to merge it back to `develop`
- [ ] RC version (https://next.storefrontcloud.io) - this is a stabilisation fix for Release Candidate of Vue Storefront. I've created branch from `release` branch and want to merge it back to `release`
- [ ] Stable version (https://demo.storefrontcloud.io) - this is an important fix for current stable version. I've created branch from `hotfix` or `master` branch and want to merge it back to `hotfix`

### Upgrade Notes and Changelog

- [x] No upgrade steps required (100% backward compatibility and no breaking changes)
- [ ] I've updated the [Upgrade notes](https://github.com/DivanteLtd/vue-storefront/blob/develop/doc/Upgrade%20notes.md) and [Changelog](https://github.com/DivanteLtd/vue-storefront/blob/develop/CHANGELOG.md) on how to port existing VS sites with this new feature

**IMPORTANT NOTICE** - Remember to update `CHANGELOG.md` with description of your change

### Contribution and currently important rules acceptance

- [x] I read and followed [contribution rules](https://github.com/DivanteLtd/vue-storefront/blob/master/CONTRIBUTING.md)
- [x] I read the [TypeScript Action Plan](https://github.com/DivanteLtd/vue-storefront/blob/master/doc/TypeScript%20Action%20Plan.md) and adjusted my PR according to it
- [x] I read about [Vue Storefront Modules](https://github.com/DivanteLtd/vue-storefront/blob/master/doc/api-modules/about-modules.md) and I am aware that every new feature should be a module

### CODE REVIEW

File: core/modules/cart/store/actions.ts [hunk 1]
```diff
@@ -275,6 +275,13 @@ const actions: ActionTree<CartState, RootState> = {
             action1: { label: i18n.t('OK') }
           })
         }
+        if (result.status === 'not_enough_stock') {
+          rootStore.dispatch('notification/spawnNotification', {
+            type: 'error',
+            message: i18n.t("not enough in stock", {name:product.name, qty:result.qty}),
```
Comment (pkarw) [inline core/modules/cart/store/actions.ts]: Cool PR - thanks @collymore. The only change I’d request is to change this i18n message. The convention still is to use the Full English translations as keys. Please just rename it and it’s fine

**Comment by pkarw** [comment]
One more thing :-) I've returned the notification after `serverUpdateItem` call. It's necessary as it may inform us about a variety of different errors (not only stock qty - however it's one of the messages that Magento item update method returns):

https://github.com/DivanteLtd/vue-storefront/commit/ef47e96fde87294c1133f847f0f8d88b858dd496

It was broken with (https://github.com/DivanteLtd/vue-storefront/commit/5ef4fadb0f34e89effcb762a33779cdae618b26d) but the situation that `out of the stock` notification isn't duplicated is ... well .. **side effect** :)

 I believe it will duplicate the out of the stock message with Your change. I suggest firing `stock/check`:
- when the user enters product page (asynchronously to not block anything and only when `isServer=false`)
- when the user changes the `qty` of product requested

So kind of `prefetch` of the qty. Then please make sure there is no duplicated `out of the stock` message anyway. 

Another solution is to just check if the return message in the `serverUpdate` is related to the stock and not display it - but it will require kind of `indexOf` on the strings ...

**Comment by collymore** [comment]
Ok thanks I'll review and work on this

**Comment by patzick** [comment]
@collymore hey Luke, hope you're doing well! Are you able to finish this PR? :)

**Comment by collymore** [comment]
Hi @patzick Im' good thanks :) hope you're well too!  I've just been so busy, I'm away this weekend so will be Tuesday before I can look at it.   If you have any time please carry on.

**Comment by pkarw** [comment]
Closing due to long inactivity; @collymore please take a look at the current develop - cart got refactored so we probably need to re-apply  Your changes

**Comment by collymore** [comment]
Ok I will do, sorry we've just been launching a site and way too many things on. We're getting back into this and I'll get this finished off on the new branch.

**Pull request closed without merging**